doc/rst/technotes/gpu.rst

```diff
@@ -112,89 +107,46 @@ Examples with multiple GPUs:
-  work; however, we only make efforts to test GPU support with this version.
+  * If using a system install of ``LLVM`` we expect this to be the same
+    version as the bundled version (currently 14). Older versions may
+    work; however, we only make efforts to test GPU support with this version.
 
 * ``CHPL_TASKS=qthreads`` is required for GPU support.
 
-* PGAS style communication is not available within GPU kernels; that is:
-  reading from or writing to a variable that is stored on a different locale
-  from inside a GPU eligible loop (when executing on a GPU) is not supported.
-
-* There is no user-level feature to specify GPU block size on a
-  per-kernel basis. This can be set on a program wide basis at compile-time by
-  passing ``--gpu-block-size=size`` to the compiler or setting it with the
-  ``CHPL_GPU_BLOCK_SIZE`` environment variable.
-
-* The use of most ``extern`` functions within a GPU eligible loop is not supported
-  (a limited set of functions used by Chapel's runtime library are supported). 
-
-   * Various functions within Chapel's standard modules call unsupported
-     ``extern`` functions and thus are not supported in GPU eligible loops.
-
-* Runtime checks such as bounds checks and nil-dereference checks are
-  automatically disabled for CHPL_LOCALE_MODEL=gpu.
-
-* For AMD GPUs:
+* ``nvcc`` (for NVIDIA) or ``hipcc`` (for AMD) must be available.
 
-    * Can only be used with local builds (i.e., CHPL_COMM=none)
+GPU-Related Environment Variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    * Certain 64-bit math functions are unsupported. To see what does
-      and doesn't work see `this test
-      <https://github.com/chapel-lang/chapel/blob/release/1.30/test/gpu/native/math.chpl>`_
-      and note which operations are executed when `excludeForRocm == true`.
+To enable GPU support set the environment variable ``CHPL_LOCALE_MODEL=gpu``
+before building Chapel.
 
-* For loops to be considered eligible for execution on a GPU they
-  must fulfill the requirements discussed in the `Overview`_ section.
+Chapel's build system will automatically try and deduce what type of GPU you
+have and where your installation of relevant runtime (e.g. CUDA or ROCM) are.
+If the type of GPU is not detected you may set ``CHPL_GPU_CODEGEN`` manually to
+either ``cuda`` (for NVIDIA GPUs) or ``rocm`` (for AMD GPUs). If the relevant
+runtime path is not automatically detected (or you would like to use a
+different installation) you may set ``CHPL_CUDA_PATH`` and/or
+``CHPL_ROCM_PATH``.
 
-* Associative arrays cannot be used on GPU sublocales with
-  ``CHPL_GPU_MEM_STRAGETY=array_on_device``.
+``CHPL_GPU_ARCH`` environment variable can be set to control the desired GPU
+architecture to compile for.  The default value is ``sm_60`` for
+``CHPL_GPU_CODEGEN=cuda`` and ``gfx906`` for ``CHPL_GPU_CODEGEN=rocm``. You may
+also use the ``--gpu-arch`` compliler flag to set GPU architecture.
```

Comment:
I know we previously mentioned how we couldn't find good documentation for what values this could be set to apart from this table in the LLVM docs:

https://llvm.org/docs/AMDGPUUsage.html#processors

But I think we're better off linking to something rather than nothing. Maybe add something like:

```
For a list of possible values see <this link to that page you found for NVIDIA> or the 
"processor" values in this table in the LLVM doc: 
https://llvm.org/docs/AMDGPUUsage.html#processors
```

And I guess as long as I'm bringing it up it's somewhat confusing that what we want forAMD is what's labeled "processor" not "architecture" in that table but the value of the environment variable is `CHPL_GPU_ARCH`. I think it's too late to rename that variable now but maybe we should create a Github issue to discuss it.

Reply:
I am not worried about "arch" at all, but I agree that the terminology there is a bit confusing. FWIW, `hipcc/clang/nvcc` all use "arch". We have `CHPL_HOST_ARCH`, `CHPL_TARGET_ARCH`. It may not be technically correct for AMD, but the precedence is very strong IMO.